sklearn/decomposition/_nmf.py

```diff
@@ -1349,7 +1378,7 @@ class NMF(_BaseNMF):
 
     Parameters
     ----------
-    n_components : int, default=None
+    n_components : int or {'auto'} or None, default=None
```

Comment:
please add a `versionchanged` directive to say that "auto" option was added in 1.4

Reply:
same for MiniBatchNMF

Reply:
done in 1c6994be8fdf7693711e5a59630101b4aee18914